---
title: "Introduction_CliniclalTranscriptLink"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_CliniclalTranscriptLink}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```


```{r setup}
devtools::install_github("michellemlin/ClinicalTranscriptLink", build_vignettes = TRUE)
library(ClinicalTranscriptLink)
```

## INTRODUCTION:

To download **ClinicalTranscriptLink**, use the following commands:

``` r
require("devtools")
devtools::install_github("michellemlin/ClinicalTranscriptLink", build_vignettes = TRUE)
library(ClinicalTranscriptLink)
```
To list all sample functions available in the package:
``` r
ls("package:ClinicalTranscriptLink")
```

## WALK THROUGH: 
Through this vignette, we will be analysing clinical and rnaseq data from The Cancer Genome Atlas Program (TCGA), specifically for the BRCA gene.

```{r}
# import datasets and save into variables
library(RTCGA.clinical)
library(RTCGA.rnaseq)

data("BRCA.rnaseq")
data("BRCA.clinical")
```
1. View datasets
```{r}
head(BRCA.rnaseq)
head(BRCA.clinical)
```

2. Extract Metadata from Clinical Dataset
```{r}
metadata <- ClinicalTranscriptLink::metadataExtract(BRCA.clinical, 
                                                    c("patient.bcr_patient_barcode",
                                                      "patient.age_at_initial_pathologic_diagnosis",
                                                      "patient.breast_carcinoma_estrogen_receptor_status",
                                                      "patient.breast_carcinoma_progesterone_receptor_status",
                                                      "patient.vital_status",
                                                      "patient.days_to_death",
                                                      "patient.race",
                                                      "patient.clinical_cqcf.histological_type"))
```
2. Merge Clinical Metadata with RNA-Seq Data
```{r}
merged <- ClinicalTranscriptLink::mergeClinicalRNA(clinical = metadata, 
                                                   RNA = BRCA.rnaseq, 
                                                   ID1 = "patient.bcr_patient_barcode", 
                                                   ID2 = "bcr_patient_barcode")
```

2. Plot a heatmap of gene expression with respect to a factor in clinical data
```{r}
heatmapGenerate(merged, "patient.race", "?|100130426")
```

3. Survival Analysis
```{r}
survivalAnalyse(metadata, timeToEvent = "patient.days_to_death", event = "patient.vital_status")
```

4. Run Shiny App
```{r}
runClinicalTranscriptLink()
```

```{r}
sessionInfo()
```
