---
title: "Introduction_CliniclalTranscriptLink"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction_CliniclalTranscriptLink}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
# devtools::install_github("michellemlin/ClinicalTranscriptLink")
library(ClinicalTranscriptLink)
```

## INTRODUCTION:

## WALK THROUGH: 
Through this vignette, we will be analysing clinical and rnaseq data from The Cancer Genome Atlas Program (TCGA), specifically for the BRCA gene.

```{r}
# import datasets and save into variables
library(RTCGA.clinical)
library(RTCGA.rnaseq)

BRCA_clinical <- BRCA.clinical
BRCA_rnaseq <- BRCA.clinical
```

1. Extract Metadata from Clinical Dataset
```{r}
metadata <- ClinicalTranscriptLink::metadataExtract(BRCA.clinical, c("patient.bcr_patient_barcode", "patient.age_at_initial_pathologic_diagnosis", "patient.breast_carcinoma_estrogen_receptor_status", "patient.breast_carcinoma_progesterone_receptor_status", "patient.clinical_cqcf.consent_or_death_status", "patient.days_to_death"))
```

2. Plot a histogram of variable of interest
```{r}
metadataDeceased <- subset(metadata, metadata$patient.clinical_cqcf.consent_or_death_status == "deceased")
metadataDeceased$patient.days_to_death <- as.numeric(metadataDeceased$patient.days_to_death)
survivalExplore(metadataDeceased, "patient.days_to_death")
```

3. Survival Analysis
```{r}
survivalMapVis(metadata, timeToEvent = "patient.days_to_death", event = "patient.clinical_cqcf.consent_or_death_status")
```

<!-- 4. Match patient IDs across the metadata and RNAseq data. This was done for this particular dataset as the IDs in RNA seq have additional codes -->
<!-- ```{r} -->
<!-- normalisedRNA <- processRNASeq(BRCA.rnaseq) -->
<!-- ``` -->
